### PR TITLE
Add ERC: Signature Validation Extension for Permit

### PR DESCRIPTION
**Pull request by yvonnezhangc**
From developers at [circlefin](https://github.com/circlefin), we are proposing a new extension for ERC-2612 permit, which has already been added to USDC v2.2. 

Discussion thread: https://ethereum-magicians.org/t/add-erc-contract-signature-validation-extension-for-eip-2612-permit/18157

### CODE REVIEW

File: ERCS/erc-TBD-permit.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
+---
+eip: TBD
```
Comment (abcoathup) [inline ERCS/erc-TBD-permit.md:2]:
```suggestion
eip: 7597
```
Assigning next sequential EIP/ERC/RIP number.
Numbering changed to sequential from 7500 and is no longer the PR number.

Please also update the filename.

**Comment by yvonnezhangc** [comment]
@abcoathup there has been some interest in having this as a standard: https://ethereum-magicians.org/t/add-erc-contract-signature-validation-extension-for-eip-2612-permit/18157

What steps do we need to take to get this merged?

**Comment by abcoathup** [comment]
@yvonnezhangc you need an ERC editor to review and address any issues. I am not an ERC editor.

**Comment by yvonnezhangc** [comment]
@xinbenlv could you help review this ERC?

File: ERCS/erc-7597.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
+---
+eip: 7597
+title: Signature Validation Extension for Permit
+description: An ERC to extend ERC-2612 Permit to support ERC-1271-based signature validation.
```
Comment (SamWilsn) [inline ERCS/erc-7597.md:4]:
```suggestion
description: Extend ERC-2612 Permit to support ERC-1271-based signature validation.
```

"An ERC" doesn't really add any useful information.

File: ERCS/erc-7597.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+discussions-to: https://ethereum-magicians.org/t/add-erc-contract-signature-validation-extension-for-eip-2612-permit/18157
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-01-15
+requires: 1271, 2612
+---
+
+# EIP: Contract signature validation extension for [ERC-2612](./eip-2612.md) Permit
+
```
Comment (SamWilsn) [inline ERCS/erc-7597.md:15]:
```suggestion
```

Title will be added by the renderer.

File: ERCS/erc-7597.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            spender,
+            value,
+            nonce,
+            deadline
+        ))
+    ));
+    
+    require(
+        // Check for both ECDSA signature and and ERC-1271 signature. A sample SignatureChecker is available at
+        // https://github.com/OpenZeppelin/openzeppelin-contracts/blob/7bd2b2a/contracts/utils/cryptography/SignatureChecker.sol
```
Comment (SamWilsn) [inline ERCS/erc-7597.md:114]:
```suggestion
        // Check for both ECDSA signature and and ERC-1271 signature.
```

Please remove the external link.

**Review by xinbenlv — APPROVED** [review]
Checked @SamWilsn 's suggested changes and seems nothing is merge-blocking.
Approving to merge. Authors please also continue to update with the suggested chagnes

**Pull request merged**